### PR TITLE
Introducing: Perltidy rule for consistent space before signatures

### PR DESCRIPTION
**Pull request by josegomezr**
Perl-tidy native rule to enforce "Consistent space before function signatures (this is not ensured by perltidy)" rule in Makefiles in openQA & os-autoinst.

### CODE REVIEW

**Review by okurz — CHANGES_REQUESTED** [review]

File: lib/perlcritic/Perl/Critic/Policy/SpaceAfterSubroutineName.pm [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+my $DESC = q{Inconsistent sub declaration};
+my $EXPL = q{Sub '%s' must have only one space surrounding name, parenthesis and/or block body.};
+
+sub default_themes { return qw(openqa) }
+sub default_severity { return $SEVERITY_HIGHEST }
+sub supported_parameters { () }
+sub applies_to { return 'PPI::Statement::Sub' }
+
+
+use Data::Dumper;
```
[the remark this answers is not in the document]
Comment (josegomezr) [reply lib/perlcritic/Perl/Critic/Policy/SpaceAfterSubroutineName.pm:26]: Nice catch

File: lib/perlcritic/Perl/Critic/Policy/SpaceAfterSubroutineName.pm [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    return $self->violation($DESC, sprintf($EXPL, $elem->name), $elem) unless $has_prototype_after_space;
+
+    my $has_space_after_prototype = $tokens[5]->isa('PPI::Token::Whitespace') && length($tokens[5]->content) == 1;
+    return $self->violation($DESC, sprintf($EXPL, $elem->name), $elem) unless $has_prototype_after_space;
+
+    my $has_block_after_prototype = $tokens[6]->isa('PPI::Token::Structure');
+
+    return $self->violation($DESC, sprintf($EXPL, $elem->name), $elem) unless $has_block_after_prototype;
+
+    return;
```
[the remark this answers is not in the document]
Comment (josegomezr) [reply lib/perlcritic/Perl/Critic/Policy/SpaceAfterSubroutineName.pm:67]: Noting

**Comment by josegomezr** [comment]
@okurz `return undef` breaks with:

```
Fatal error while critiquing "lib/perlcritic/Perl/Critic/Policy/ArgumentInUseStrictWarnings.pm": Can't call method "location" on an undefined value at /home/josegomezr/perl5/lib/perl5/Perl/Critic.pm line 175.
```

I'll revert it.

**Comment by josegomezr** [comment]
I believe the latest rework @ [f6971a0](https://github.com/os-autoinst/os-autoinst-common/pull/35/commits/f6971a00ef3eb721a96e2b3786aebb2ce6463cec) makes it more straight forward

**Review by okurz — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Comment by perlpunk** [comment]
btw, this PR talks about perltidy, but it is about perlcritic.

**Comment by josegomezr** [comment]
Ouh, just noticed it :sweat: I'll update the title although so long for the commit :sweat_smile:

**Pull request merged**